packages/block-library/src/search/index.php

```diff
@@ -203,11 +205,56 @@ function styles_for_block_core_search( $attributes ) {
 			$default_padding  = 4;
 			$adjusted_radius  = $border_radius + $default_padding;
 			$wrapper_styles[] = sprintf( 'border-radius: %dpx;', esc_attr( $adjusted_radius ) );
+			$input_styles[]   = sprintf( 'border-radius: %dpx;', esc_attr( $adjusted_radius ) );
+		}
+	}
+
+	// Add color styles.
+	$has_colors = ! empty( $attributes['style']['color'] );
+
+	if ( $has_colors ) {
+		if ( $attributes['style']['color']['text'] ) {
+			$button_styles[] = sprintf( 'color: %s;', esc_attr( $attributes['style']['color']['text'] ) );
+		}
+		if ( $attributes['style']['color']['background'] ) {
+			$button_styles[] = sprintf( 'background-color: %s;', esc_attr( $attributes['style']['color']['background'] ) );
 		}
 	}
 
 	return array(
-		'shared'  => ! empty( $shared_styles ) ? sprintf( ' style="%s"', implode( ' ', $shared_styles ) ) : '',
+		'input'   => ! empty( $input_styles ) ? sprintf( ' style="%s"', implode( ' ', $input_styles ) ) : '',
+		'button'  => ! empty( $button_styles ) ? sprintf( ' style="%s"', implode( ' ', $button_styles ) ) : '',
 		'wrapper' => ! empty( $wrapper_styles ) ? sprintf( ' style="%s"', implode( ' ', $wrapper_styles ) ) : '',
 	);
 }
+
+/**
+ * Returns color classnames depending on whether there are named or custom border colors.
+ *
+ * @param array $attributes The block attributes.
+ *
+ * @return string The color classnames to be applied to the block elements.
+ */
+function get_color_classes_for_block_core_search( $attributes ) {
+	$classnames = array();
+
+	// Text color.
+	$has_custom_text_color = ! empty( $attributes['style']['color']['text'] );
+	if ( ! empty( $attributes['textColor'] ) ) {
```

Comment:
Style choice: can we have both `$has_custom_text_color` and `$has_preset_text_color`? Same for background below.

Reply:
Done! 👍